### PR TITLE
5118-dce---fix-aws-nuke-for-awsbackup

### PR DESCRIPTION
**Pull request by Suraj-Singh-hooda**

### CODE REVIEW

File: resources/backup-vaults-access-policies.go [hunk 1]
```diff
@@ -117,3 +120,10 @@ func (b *BackupVaultAccessPolicy) Remove() error {
 func (b *BackupVaultAccessPolicy) String() string {
 	return b.backupVaultName
 }
+
+func (b *BackupVaultAccessPolicy) Filter() error {
+	if strings.HasPrefix(b.backupVaultName, "aws/efs/automatic-backup-vault") {
+		return fmt.Errorf("cannot delete EFS automatic backups backup policy")
```
Comment (bjfish25) [inline resources/backup-vaults-access-policies.go:126]: If we return an error here does it fail the nuke? If so should return nil

Comment (Suraj-Singh-hooda) [reply resources/backup-vaults-access-policies.go:126]: ok made that nil, thanks

File: resources/backup-vaults-access-policies.go [hunk 2]
```diff
@@ -117,3 +119,10 @@ func (b *BackupVaultAccessPolicy) Remove() error {
 func (b *BackupVaultAccessPolicy) String() string {
 	return b.backupVaultName
 }
+
+func (b *BackupVaultAccessPolicy) Filter() error {
+	if strings.HasPrefix(b.backupVaultName, "aws/efs/automatic-backup-vault") {
+		return nil
```
Comment (bjfish25) [inline resources/backup-vaults-access-policies.go]: Sorry just checked other resources. I do think we want the fmt.Error here

Comment (Suraj-Singh-hooda) [reply resources/backup-vaults-access-policies.go]: no issue, thanks for quick review

**Review by bjfish25 — APPROVED** [review]

**Pull request merged**